proteinworkshop/datasets/pdb_dataset.py

```diff
@@ -30,9 +31,10 @@ def __init__(
         remove_non_standard_residues: bool,
         remove_pdb_unavailable: bool,
         train_val_test: List[float],
-        split_type: Literal["sequence_similarity", "random"],
+        split_type: Literal["sequence_similarity", "time_cutoff", "random"],
         split_sequence_similiarity: int,
-        overwrite_sequence_clusters: bool
+        overwrite_sequence_clusters: bool,
+        split_time_frames: List[str]
```

Comment:
Should be `Optional[List[str]]`. This arg probs needs a check for datetime format.

Reply:
In theory it is optional, but we cannot specify a default option as it is later in the argument list. If we want to make the type hint optional, should we move it up the list and give it a default value?

Reply:
Hmm, I'm generally in favour of fewer args (and not passing in things that aren't used since it can be confusing). What do you think about a pattern where we can pass the time splits into `train_test_split` instead of the list of floats and the behaviour is controlled by `split_type` (with appropriate error catching)?

Reply:
cc @amorehead

Reply:
With `pass the time splits into train_test_split`, do you mean giving tuples of (split_ratio, split_time_cutoff) if the time version is chosen for example? And if the time version is not chosen, one would just take the first element of that tuple, with the default for the second being None?

Reply:
Yeah, I think clarification would be helpful here

Reply:
Was not sure what exactly you meant @a-r-j, but for now, I just reordered arguments so that we can give defaults to these and the user does not need to specify them. Does that work for you?

Reply:
@a-r-j what do you think?